src/applications/mhv/secure-messaging/tests/components/Dashboard/DashboardUnreadMessages.unit.spec.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { expect } from 'chai';
+import { renderWithStoreAndRouter } from '@department-of-veterans-affairs/platform-testing/react-testing-library-helpers';
+import reducer from '../../../reducers';
+import DashboardUnreadMessages from '../../../components/Dashboard/DashboardUnreadMessages';
+import { Paths, ErrorMessages } from '../../../util/constants';
+import inbox from '../../fixtures/folder-inbox-metadata.json';
+
+describe('DashboardUnreadMessages component', () => {
+  const setup = folder => {
+    return renderWithStoreAndRouter(
+      <DashboardUnreadMessages inbox={folder} />,
+      {
+        initialState: {},
+        reducers: reducer,
+      },
+    );
+  };
+
+  it('should not render if inbox is undefined', () => {
+    const { container } = setup(undefined);
+    expect(container).to.be.empty;
+  });
+
+  it('should render error message if inbox is null', () => {
+    const { getByText } = setup(null);
+    expect(getByText(ErrorMessages.LandingPage.GET_INBOX_ERROR)).to.exist;
+
+    expect(getByText('Go to your inbox')).to.have.attribute(
+      'href',
+      Paths.INBOX,
+    );
+    expect(getByText('Start a new message')).to.have.attribute(
+      'href',
+      Paths.COMPOSE,
+    );
+  });
+
+  it('should render unread message count if inbox is defined', () => {
+    const customInbox = { ...inbox, unreadCount: 15 };
+    const { getByText } = setup(customInbox);
+    expect(
+      getByText(`${customInbox.unreadCount} unread messages in your inbox`),
+    ).to.exist;
+    expect(getByText('Go to your inbox')).to.have.attribute(
+      'href',
+      Paths.INBOX,
+    );
+    expect(getByText('Start a new message')).to.have.attribute(
+      'href',
+      Paths.COMPOSE,
+    );
+  });
+
+  it('should render singular unread message count if inbox is defined and unreadCount is 1', () => {
+    const customInbox = { ...inbox, unreadCount: 1 };
+    const { getByText } = setup(customInbox);
+    expect(getByText(`${customInbox.unreadCount} unread message in your inbox`))
+      .to.exist;
+  });
```

Comment:
This test is great -- I just have a question about if it ever possible for unreadCount to be undefined or null? If it's a possibility can we add a test that prints what the expected text response should be ? (ie. '1 unread message in your inbox' instead of 'undefined unread message in your inbox'.  If not then disregard😶‍🌫️

Reply:
We are getting `null` if the response did not return back `inbox` folder object. In that case, we will get `Unable to retrieve messages` error in that line. I am handling `undefined` to not render a component so that Error not flashing before we get a response from API. So that way we are excluding a possibility of seeing `undefined unread messages in your inbox`. Great question!

Reply:
there is a test above that handles both `undefined` and `null` value scenarios